src/Files.App/Services/Windows/WindowsQuickAccessService.cs

```diff
@@ -26,7 +26,12 @@ public async Task<IEnumerable<ShellFileItem>> GetPinnedFoldersAsync()
 		private async Task PinToSidebarAsync(string[] folderPaths, bool doUpdateQuickAccessWidget)
 		{
 			foreach (string folderPath in folderPaths)
-				await ContextMenu.InvokeVerb("pintohome", [folderPath]);
+			{
+				// make sure that the item has not yet been pinned
+				// the verb 'pintohome' is for both adding and removing
+				if (!IsItemPinned(folderPath))
```

Comment:
I suppose this doesn't hurt, but from what I understand, the changes to `QuickAccessManager` should be enough to resolve the issue.

Reply:
That's correct.
I just thought it would be good to add this check here to prevent problems like this. 

Since the method is called 'PinToSidebarAsync', someone using the method would not assume that the item gets unpinned if it is already pinned.